packages/state-transition/src/cache/epochCache.ts

```diff
@@ -383,16 +422,46 @@ export class EpochCache {
       throw Error("totalActiveBalanceIncrements >= Number.MAX_SAFE_INTEGER. MAX_EFFECTIVE_BALANCE is too low.");
     }
 
-    const currentShuffling =
-      cachedCurrentShuffling ??
-      computeEpochShuffling(state, currentActiveIndices, currentActiveIndices.length, currentEpoch);
-    const previousShuffling =
-      cachedPreviousShuffling ??
-      (isGenesis
-        ? currentShuffling
-        : computeEpochShuffling(state, previousActiveIndices, previousActiveIndices.length, previousEpoch));
-    const nextShuffling =
-      cachedNextShuffling ?? computeEpochShuffling(state, nextActiveIndices, nextActiveIndices.length, nextEpoch);
+    const nextActiveIndices = new Uint32Array(nextActiveIndicesAsNumberArray);
+    let previousShuffling: EpochShuffling;
+    let currentShuffling: EpochShuffling;
+    let nextShuffling: EpochShuffling;
+
+    {
+      const previousActiveIndices = new Uint32Array(previousActiveIndicesAsNumberArray);
+      const currentActiveIndices = new Uint32Array(currentActiveIndicesAsNumberArray);
+      if (!shufflingCache) {
+        // Only for testing. shufflingCache should always be available in prod
+        previousShuffling = computeEpochShuffling(state, previousActiveIndices, previousEpoch);
+        currentShuffling = isGenesis
+          ? previousShuffling
+          : computeEpochShuffling(state, currentActiveIndices, currentEpoch);
+        nextShuffling = computeEpochShuffling(state, nextActiveIndices, nextEpoch);
+      } else {
+        currentShuffling = cachedCurrentShuffling
+          ? cachedCurrentShuffling
+          : shufflingCache.getSync(currentEpoch, currentDecisionRoot, {
+              state,
+              activeIndices: currentActiveIndices,
```

Comment:
this happens when we reload state. It's more likely than not we have cached shuffling already
```suggestion
              activeIndices: new Uint32Array(currentActiveIndicesAsNumberArray),
```
no need to create `currentActiveIndices`

Reply:
same to `previousShuffling` and `nextShuffling`

Reply:
ok.  refactored